Makefile

```diff
@@ -1,20 +1,21 @@
 .PHONY: build push run
 
 ECR_REGISTRY := 723255503624.dkr.ecr.us-east-1.amazonaws.com
-PROJECT_NAME := ship-it
 VERSION := $(shell git rev-parse HEAD)
 
-IMAGE := $(PROJECT_NAME):$(VERSION)
-LATEST_IMAGE := $(PROJECT_NAME):latest
+env-target:
+ifndef TARGET
+	    $(error TARGET is undefined)
+endif
 
-build:
-	docker build -t $(IMAGE) .
+build: env-target
```

Comment:
Is there a better way to manage building separate targets? This works, but it becomes awkward to have a single "build all the things" target for travis.

Reply:
Yeah I don't think there's much better we can do with Makefiles, I think this is OK for now.